spyder/plugins/maininterpreter/confpage.py

```diff
@@ -117,6 +118,30 @@ def setup_page(self):
         self.def_exec_radio.toggled.connect(self.cus_exec_combo.setDisabled)
         self.cus_exec_radio.toggled.connect(self.cus_exec_combo.setEnabled)
         pyexec_layout.addWidget(self.cus_exec_combo)
+
+        system_pythonpath = newcb(
+            _("Use system PYTHONPATH"),
+            'system_pythonpath',
+            msg_info=_("Please note that these changes will "
+                       "be applied only to new consoles but "
+                       "and immediately to completions"),
+            tip=_("For standalone applications, use your system PYTHONPATH")
+        )
+        pyexec_layout.addWidget(system_pythonpath)
+        system_env_variables = newcb(
+            _("Use system environment variables"),
+            'system_env_variables',
+            msg_info=_("Please note that these changes will "
+                       "be applied only to new consoles"),
+            tip=_("For standalone applications, use your "
+                  "system environment variables")
+        )
+        pyexec_layout.addWidget(system_env_variables)
+        if not running_in_mac_app() and not is_pynsist():

```

Comment:
I think it'd be nice to have this on Linux because the usage of env vars is quite common there.

Reply:
What purpose will the widget serve when launching Spyder from a non-app installation? When launching from commandline, the user environment variables will already be included, yes? If you want to keep the addition to the preferences for non-app installations, then deselecting `system_env_variables` in preferences would require _removing_ user environment variables which would be very tricky.

Reply:
In fact, it looks like even the Windows standalone application already includes the user's environment variables. So this is only relevant to the macOS app.

Reply:
> What purpose will the widget serve when launching Spyder from a non-app installation?

You can launch Spyder graphically from the Anaconda Navigator or the equivalent to the Windows menu. So, in those cases it's very useful to pass your env vars to the kernel.

> So this is only relevant to the macOS app.

Great! Then, I suggest making this work for Unix systems (because Spyder can also be launched graphically on Mac when not using our installer).

Reply:
Okay, I'm confused.
* Launching Spyder either from the commandline or graphically (Anaconda Navigator) on all platforms will already inherit the user's environment variables, including PYTHONPATH.
* Additionally, launching Spyder's standalone application on Windows platform will already inherit the user's environment variables, including PYTHONPATH.
* Only launching Spyder's standalone application on macOS platform will NOT inherit the user's environment variables, nor the user's system PYTHONPATH.

I'm not sure what you mean by "making this work for Unix systems".

In light of the above, we have the following possibilities:
1. Eliminate the preference altogether and always use the user's environment variables (including PYTHONPATH) when using the standalone application on macOS.
2. Provide the preference ONLY for the standalone application on macOS.
3. Provide the preference for all launching scenarios.

Options 1 and 2 are easy. Option 3 can be done but involves _removing_ environment variables when the option is deselected; this will be true for unix systems as well as Windows. Did you mean to support option 3?